sycl/include/sycl/types.hpp

```diff
@@ -21,6 +21,9 @@
 #include <sycl/half_type.hpp>                  // for StorageT, half, Vec16...
 #include <sycl/marray.hpp>                     // for __SYCL_BINOP, __SYCL_...
 #include <sycl/multi_ptr.hpp>                  // for multi_ptr
-#include <sycl/vector.hpp>                     // for sycl::vec and swizzles
+
+#include <sycl/vector.hpp>
+
+#include <sycl/detail/vector_convert.hpp>
```

Comment:
Why aren't we including `sycl/detail/vector_convert.hpp` inside `vector.hpp`? `vec::convert<>` is a member function of `sycl::vec`, so IMO, it should go in latter.

Reply:
That would undo all the benefits mentioned in the description. Having it separate and optional is exactly what I'm trying to do here.

Reply:
```
It keeps all convert-related code in one place
```
This would still be the case. I understand the rationale of moving the body of `vec::convert<>`  inside the `vector_convert.hpp` header. What I'm not convinced about is whether we should include `vector_convert.hpp` in `vector.hpp` or not.

```
It allows not to include that extra functionality if this particular
method isn't used
```
What benefit will this bring? `vector_convert.hpp` is a "relatively" small header consisting mainly of compile-time MACROs and a few functions. Do we see any reduction in compile-time if we not include this header in `vector.hpp`?

Reply:
It needs `half` and `bfloat16` to be complete, while the rest can work just fine with incomplete forward declaration.

Reply:
> It needs half and bfloat16 to be complete

Ahh.. I see. Makes sense now.